src/adapter/src/subscribe.rs

```diff
@@ -316,12 +369,38 @@ impl ActiveSubscribe {
 
 /// The reason for removing an [`ActiveSubscribe`].
 #[derive(Debug, Clone)]
-pub enum SubscribeRemovalReason {
-    /// The subscribe completed successfully.
+pub enum ComputeSinkRemovalReason {
+    /// The compute sink completed successfully.
     Finished,
-    /// The subscribe was canceled due to a user request.
+    /// The compute sink was canceled due to a user request.
     Canceled,
-    /// The subscribe was forcibly terminated because an object it depended on
+    /// The compute sink was forcibly terminated because an object it depended on
     /// was dropped.
     DependencyDropped(String),
 }
+
+/// A description of an active copy to from coord's perspective.
+#[derive(Debug)]
+pub struct ActiveCopyTo {
+    /// The user of the session that created the subscribe.
+    pub user: User,
+    /// The connection id of the session that created the subscribe.
+    pub conn_id: ConnectionId,
+    /// The context about the COPY TO statement getting executed.
+    /// Used to eventually call `ctx.retire` on.
+    pub ctx: Option<ExecuteContext>,
```

Comment:
Very scared about having this be an option, because it obfuscates the responsibility of which component owns ctx retiring.

Reply:
This is my fault, because I changed subscribe to use an `Option` in the morally equivalent spot: https://github.com/MaterializeInc/materialize/pull/25179/files#diff-20ea8c38b1da4ccc88fd84efcf5a99097e62bc1811e726cee5562c9c97c4bb98R41

The idea is that `None` indicates "final response already sent to the subscribe (or copy to), move along, all that's remaining is to call `drop_collections` on the thing."

Reply:
(Resolving since we addressed this in slack)